src/features/authentication/forgot_password/forgotPassword.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { useEffect, useState } from "react";
+import { sendPasswordResetEmail } from "firebase/auth";
+import { useNavigate } from "react-router-dom";
+import { InputText } from "primereact/inputtext";
+import { Button } from "primereact/button";
+
+import logo from "../../../common/assets/Logo.png";
+import "./forgotPassword.scss";
+
+import { observer } from "mobx-react-lite";
+import { auth } from "../../../config/firebase-config";
+
+export const ForgotPasswordForm: React.FC = observer(() => {
+  const navigate = useNavigate();
+
+  const [isSubmitPress, setIsSubmitPress] = useState(false);
+  const [successfulSubmit, setSuccessfulSubmit] = useState(false);
+
+  const initialValues = {
+    email: "",
+    password: "",
+  };
+  const [formErrors, setFormErrors] = useState(initialValues);
+  const [email, setEmail] = useState("");
+
+  const forgotPassword = (email: string) => {
+    return sendPasswordResetEmail(auth, email);
+  };
+
+  const validate = (value: any) => {
+    const errors = initialValues;
+
+    const regex = /^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$/i;
+    if (!value) {
+      errors.email = "Email is required.";
+    } else if (!regex.test(value)) {
+      errors.email = "This is not a valid email format!";
+    }
+    return errors;
+  };
+
+  const handleSubmit = (e: any) => {
+    e.preventDefault();
+    setFormErrors(validate(email));
+    setIsSubmitPress(true);
+  };
+
+  useEffect(() => {
+    const errors = initialValues;
+    // validating form errors to see if we can loggin to app
+    if (!formErrors.email.length && isSubmitPress) {
+      forgotPassword(email)
+        .then(() => {
+          console.log("Forgot password success");
+          setSuccessfulSubmit(true);
+        })
+        .catch((error) => {
+          console.log(error.message);
+          if (error.message.includes("user-not-found")) {
+            errors.email = "Email is incorrect.";
+            setFormErrors(errors);
+          }
+          setSuccessfulSubmit(false);
+        });
+    }
+  }, [formErrors]);
```

Comment:
I mean, normally, it shouldn't work if there is no "isSubmitPress" in the dependency array, in the case there is no errors, because I didn't mutate any state for the formErrors, but I see that for me it is not checking that, or idk. It sees the "setFormsErrors()" action and then the useEffect is getting ready for ground and pound :))

Strange thing this React.